### PR TITLE
Explicitly enable the foreman dnf module on Foreman 3.2+

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: manifests/repo.pp [hunk 1]
```diff
@@ -41,6 +41,14 @@
         provider    => 'dnfmodule',
       }
     }
+
+    if $facts['os']['release']['major'] == '8' and ($repo == 'nightly' or versioncmp($repo, '3.2') >= 0) {
+      package { 'foreman':
+        ensure      => 'latest',
```
Comment (evgeni) [inline manifests/repo.pp:47]: maybe "latest" wasn't that a good name after all? :D

File: manifests/repo.pp [hunk 2]
```diff
@@ -41,6 +41,14 @@
         provider    => 'dnfmodule',
       }
     }
+
+    if $facts['os']['release']['major'] == '8' and ($repo == 'nightly' or versioncmp($repo, '3.2') >= 0) {
```
Comment (ekohl) [inline manifests/repo.pp:45]: With this, is it still needed to enable Ruby? Or should this be first and then the `>= 2.5` as an `elsif`.

Comment (evgeni) [reply manifests/repo.pp:45]: yea, the >=2.5 can be an eslif

File: manifests/repo.pp [hunk 3]
```diff
@@ -34,7 +34,13 @@
       Foreman::Repos['foreman'] -> Package['centos-release-scl-rh']
     }
 
-    if $facts['os']['release']['major'] == '8' and ($repo == 'nightly' or versioncmp($repo, '2.5') >= 0) {
+    if $facts['os']['release']['major'] == '8' and ($repo == 'nightly' or versioncmp($repo, '3.2') >= 0) {
+      package { 'foreman:latest':
```
Comment (evgeni) [inline manifests/repo.pp:38]: yes, this essentially breaks the fix from #1015, but I need to see if it works before we rename the stream on `yum.tfm`

**Comment by evgeni** [comment]
this should not be merged before we didn't get a decision what we rename `latest` to in https://github.com/theforeman/foreman-packaging/pull/7522

File: manifests/repo.pp [hunk 4]
```diff
@@ -34,7 +34,14 @@
       Foreman::Repos['foreman'] -> Package['centos-release-scl-rh']
     }
 
-    if $facts['os']['release']['major'] == '8' and ($repo == 'nightly' or versioncmp($repo, '2.5') >= 0) {
+    if $facts['os']['release']['major'] == '8' and ($repo == 'nightly' or versioncmp($repo, '3.2') >= 0) {
+      package { 'foreman':
```
Comment (evgeni) [inline manifests/repo.pp:38]: funny how we nowhere have a `package { 'foreman' }` in other places of the code…

Comment (ekohl) [reply manifests/repo.pp:38]: That also surprised me.

**Comment by evgeni** [comment]
of course, this now also fails to install hammer_cli_katello, because the katello module ain't enabled.

**Review by ekohl — APPROVED** [review]
I think we'll also need a similar change in [katello::repo](https://github.com/theforeman/puppet-katello/blob/master/manifests/repo.pp).

**Pull request merged**